io/io/inc/ROOT/RIoUring.hxx

```diff
@@ -0,0 +1,52 @@
+/*************************************************************************
+ * Copyright (C) 1995-2020, Rene Brun and Fons Rademakers.               *
+ * All rights reserved.                                                  *
+ *                                                                       *
+ * For the licensing terms see $ROOTSYS/LICENSE.                         *
+ * For the list of contributors see $ROOTSYS/README/CREDITS.             *
+ *************************************************************************/
+
+#ifndef ROOT_RIoUring
+#define ROOT_RIoUring
+
+#include <liburing.h>
+#include <liburing/io_uring.h>
+
+#include <ROOT/RError.hxx>
+
+namespace ROOT {
+namespace Internal {
+
+class RIoUring {
+private:
+   struct io_uring fRing;
+public:
+   explicit RIoUring(size_t size) {
+      int ret = io_uring_queue_init(size, &fRing, 0 /* no flags */);
+      if (ret) {
+         throw R__FAIL("couldn't open ring");
+      }
+   }
+
+   RIoUring(const RIoUring&) = delete;
+   RIoUring& operator=(const RIoUring&) = delete;
+
+   ~RIoUring() {
+      io_uring_queue_exit(&fRing);
+   }
+
+   /// Check if io_uring is available on this system.
+   static bool IsAvailable() {
+      try {
+         RIoUring(1);
+      } catch (const Experimental::RException& _err) {
```

Comment:
Simply `catch (const Experimental::RException&)` ?

Reply:
Thank you, see e63ffc1